src/openbench/metric/keyword_boosting_metrics/boosting_metrics.py

```diff
@@ -0,0 +1,265 @@
+import os
+import json
+from typing import List, Dict, Any
+import texterrors
+import logging
+from pathlib import Path
+
+from pyannote.metrics.base import BaseMetric
+from ..registry import MetricRegistry
+from ..metric import MetricOptions
+from ...types import PipelineType
+from ...pipeline_prediction import Transcript
+
+
+
+
+
+class BaseKeywordMetric(BaseMetric):
+    """Base class for keyword boosting metrics."""
+    
+    def __init__(self):
+        """Initialize keyword metric."""
+        super().__init__()
+        # Use Whisper's BasicTextNormalizer
+        from transformers.models.whisper.english_normalizer import BasicTextNormalizer
+        self.text_normalizer = BasicTextNormalizer()
+
+    def compute_keyword_stats(
+        self, reference: Transcript, hypothesis: Transcript, **kwargs
+    ) -> Dict[str, Any]:
+        """Compute keyword statistics between reference and hypothesis."""
+        # Get keywords from kwargs (passed by the metric framework)
+        keywords = kwargs.get('dictionary', [])
+        
+        # Convert transcripts to text
+        ref_text = " ".join([word.word for word in reference.words])
+        hyp_text = " ".join([word.word for word in hypothesis.words])
+        
+        print(f"Reference text: '{ref_text}'")
+        print(f"Hypothesis text: '{hyp_text}'")
+        print(f"Keywords: {keywords}")
+        
+        # Apply normalization to BOTH reference and hypothesis
+        ref_text = self.text_normalizer(ref_text)
+        hyp_text = self.text_normalizer(hyp_text)
+        
+        # Normalize keywords as well
+        normalized_keywords = [self.text_normalizer(kw) for kw in keywords]
+        
+        print(f"Normalized Reference: '{ref_text}'")
+        print(f"Normalized Hypothesis: '{hyp_text}'")
+        print(f"Normalized Keywords: {normalized_keywords}")
+        
+        # Get alignment using texterrors
+        ref_words = ref_text.split()
+        hyp_words = hyp_text.split()
+        texterrors_ali = texterrors.align_texts(ref_words, hyp_words, False)
+        
+        # Create alignment pairs
```

Comment:
Is there a code reference link if there are chunks of copied code from NeMo?

Reply:
Also, we could benefit from extended code comments to explain the next ~100 lines